tket/include/tket/PauliGraphRefactor/PGOp.hpp

```diff
@@ -162,6 +198,12 @@ class PGOp {
    */
   virtual std::vector<SpPauliStabiliser> active_paulis() const = 0;
 
+  /**
+   * A structured version of active_paulis() which groups the Pauli operators
+   * according to their commutativity with each other.
+   */
+  virtual PGOp_signature pauli_signature() const = 0;
```

Comment:
`active_paulis` is still convenient for when I want one iterable container to go through all of them rather than having to unpack the structured signature. Similarly, there is an argument it duplicates `port` as having a means to access all of the active Pauli operators in a manner independent of the subclass, but the vector is just convenient for coding with